### PR TITLE
[RealImage/apsara #1163] Fix compiler warnings

### PR DESCRIPTION
**Pull request by VValkovoy**
# What this PR does?
* Fixes playback of `gst-plugin-imbsink` Gstreamer plugin regarding compiler warnings treated as errors.

- **Issue**
 Currently ImbSink plugin has multiple warnings during building treated as errors by compiler.

- **Solution**
    * Fix compiler`s build warnings

- **Expected behavior**
`gst-plugin-imbsink` Gstreamer plugin should be built wothout any warnings.

# How to build:
* `git clone --recursive -b feature/fix-warnings git@github.com:RealImage/gst-plugin-imbsink.git`
* `cmake  ..`
* `make -j4`

### CODE REVIEW

File: src/BufferQueue.cpp [hunk 1]
```diff
@@ -315,6 +278,36 @@ MediaBlock::BufferQueue::BufferingStatus MediaBlock::BufferQueue::GetBufferStatu
     return buffering_status.load();
 }
 
+/* Peek for the next buffer */
+bool MediaBlock::BufferQueue::PeekGstBuffer(GstBuffer*& buffer)
```
Comment (Iyer-VivekAnand) [inline src/BufferQueue.cpp]: _ PeekGstBuffer ?

File: src/BufferQueue.cpp [hunk 2]
```diff
@@ -315,6 +278,36 @@ MediaBlock::BufferQueue::BufferingStatus MediaBlock::BufferQueue::GetBufferStatu
...
+{
+    g_mutex_lock(&buffer_lock);
+    while (buffer_queue->empty())
+    {
+        GST_DEBUG("Buffer Queue is empty. Queue size is [%lu]", buffer_queue->size());
```
Comment (Iyer-VivekAnand) [inline src/BufferQueue.cpp]: Would be good if we log this once as warning and the time it takes to recover.

Comment (VValkovoy) [reply src/BufferQueue.cpp]: Sure, @Iyer-VivekAnand. Will do it in a separate PR on a top priority.

File: src/BufferQueue.cpp (hunk 1 above)
Comment (VValkovoy) [reply src/BufferQueue.cpp]: Done.

File: src/BufferQueue.cpp (hunk 2 above)
Comment (VValkovoy) [reply src/BufferQueue.cpp]: @Iyer-VivekAnand  
Done in current PR.

File: src/BufferQueue.cpp [hunk 3]
```diff
@@ -248,27 +223,78 @@ void MediaBlock::BufferQueue::TerminateBuffering()
...
+{
+    return buffering_status.load();
+}
+
+/* Peek for the next buffer */
+bool MediaBlock::BufferQueue::_PeekGstBuffer(GstBuffer*& buffer)
+{
+    buffer = nullptr;
+
+    g_mutex_lock(&buffer_lock);
```
[the remark this answers is not in the document]
Comment (VValkovoy) [reply src/BufferQueue.cpp]: Moved.

File: src/BufferQueue.cpp [hunk 4]
```diff
@@ -248,27 +223,78 @@ void MediaBlock::BufferQueue::TerminateBuffering()
...
+
+    const gint64 WAIT_TIME_SEC = 3;
+    gint64       end_time = g_get_monotonic_time() + WAIT_TIME_SEC * G_TIME_SPAN_SECOND;
```
[the remark this answers is not in the document]
Comment (VValkovoy) [reply src/BufferQueue.cpp]: Added.

File: src/BufferQueue.cpp [hunk 5]
```diff
@@ -278,41 +304,24 @@ bool MediaBlock::VideoBufferQueue::Peek(Buffer& mediaBuffer)
...
-bool MediaBlock::AudioBufferQueue::Peek(Buffer& mediaBuffer)
+bool MediaBlock::AudioBufferQueue::Peek(MediaBlock::Buffer& mediaBuffer)
 {
-    GST_DEBUG("Peeking Audio buffer");
+    GST_DEBUG("Peeking Audio buffer from queue. Queue size is [%lu].", buffer_queue->size());
     GstBuffer* buffer = nullptr;
 
-    bool ret = BufferQueue::Peek(buffer);
-    if (ret && buffer)
+    bool result = _PeekGstBuffer(buffer);
```
[the remark this answers is not in the document]
Comment (VValkovoy) [reply src/BufferQueue.cpp]: Added.

**Pull request merged**